biogeochem/EDMortalityFunctionsMod.F90

```diff
@@ -166,24 +169,44 @@ subroutine mortality_rates( cohort_in,bc_in,btran_ft, mean_temp,             &
        ! Carbon Starvation induced mortality.
        if ( cohort_in%dbh  >  0._r8 ) then
 
-          ! We compare storage with leaf biomass if plant were fully flushed, otherwise
-          ! mortality would be underestimated for plants that lost all leaves and have no
-          ! storage to flush new ones.
-          ! MLO. Why isn't this comparing with storage allometry (i.e., accounting for
-          !      cushion)?
+          ! Find the current ratio between storage biomass and leaf biomass, which will be
+          ! used to define carbon starvation mortality.  The reference leaf biomass is 
+          ! always for when plants are fully flushed (but accounting for damage and 
+          ! trimming).
           call bleaf(cohort_in%dbh,cohort_in%pft,cohort_in%crowndamage,cohort_in%canopy_trim, &
                1.0_r8, target_leaf_c)
           store_c = cohort_in%prt%GetState(store_organ,carbon12_element)
-
           call storage_fraction_of_target(target_leaf_c, store_c, frac)
-          if( frac .lt. 1._r8) then
-             cmort = max(0.0_r8,EDPftvarcon_inst%mort_scalar_cstarvation(cohort_in%pft) * &
-                  (1.0_r8 - frac))
-          else
+
+          ! Select the carbon starvation mortality model (linear or exponential)s.
+          select case (mort_cstarvation_model)
+          case (cstarvation_model_lin)
+             ! Linear model. Carbon starvation mortality will be zero when fraction of
+             ! storage is greater than or equal to mort_upthresh_cstarvation, and will
+             ! increase to the maximum mortality (mort_scalar_cstarvation) when frac = 0.
+             cmort = EDPftvarcon_inst%mort_scalar_cstarvation(cohort_in%pft) * &
+                max(0.0_r8, (EDPftvarcon_inst%mort_upthresh_cstarvation(cohort_in%pft)-frac) / &
```

Comment:
I see EDPftvarcon_inst%mort_upthresh_cstarvation is already in the parameter file, that makes things a little easier: https://github.com/NGEET/fates/blob/sci.1.70.0_api.32.0.0/main/EDPftvarcon.F90#L1030-L1032 with a default of 1, which should give us b4b with base on existing tests.

Reply:
Thinking about how this upthresh parameter will affect results. So if you do set it lower than 1, then cstarvation won't kick in at all, until the plants storage fraction hits that upthresh fraction... but since we now divide by uptresh as well, the rate at which cstarvation ramps up with be increased once it does, so it should converge to the same rate as the storage fraction approaches zero.

Reply:
Correct, this is just to make the existing code general and continuous for any value of `mort_upthresh_cstarvation`, whilst keeping the solution identical to the previous code when `mort_upthresh_cstarvation = 1.0`